### ISSUE TITLE
MySql5Processor does not allow to provide custom IDbFactory

### ISSUE THREAD

**Issue by rhegner** [fluentmigrator/fluentmigrator#934]
The constructor of `MySql5Processor` requests a concrete instance of `MySqlDbFactory` from DI. This makes it hard to provide a custom DbFactory. It forces me to derive my custom DbFactory from `MySqlDbFactory`, which I don't want.

I suggest that the constructor of `MySql5Processor` should request a more generic type (ideally this would be `IDbFactory` or some other pure interface - `DbFactoryBase` would also be acceptable). This would allow me to provide an implementation which is not derived from `MySqlDbFactory` or `ReflectionBasedDbFactory`.

Background: In my project I'd like to use `MySqlConnector` instead of `MySql.Data`. My current workaround is to derive from `MySqlDbFactory`

    private class MySqlConnectorDbFactory : MySqlDbFactory
    {
        public MySqlConnectorDbFactory(IServiceProvider serviceProvider)
            : base(serviceProvider)
        { }

        protected override DbProviderFactory CreateFactory()
            => MySqlClientFactory.Instance;
    }
 and then to configure the builder like this:

    builder.Services.AddScoped<MySqlDbFactory, MySqlConnectorDbFactory>();
    builder.AddMySql5();

But I don't think this is a particularly nice solution.

**Comment by fubar-coder** [comment]
I see where your problem is. The reason why we cannot use `IDbFactory` is that we need to support configurations for multiple databases (and connections) and the ability to switch between them on the fly. Maybe you can come up with a better solution that supports this scenario?

**Comment by fubar-coder** [comment]
Hmm, yeah, an interface hierarchy would solve the problem. 🤔

**Comment by rhegner** [comment]
Yeah would be great if we could have that in some future version, so that I can clean up my code.

In the meantime - does my workaround look correct, or do I need to override any other members (there are a couple of obsolete members there)? Based on my first quick tests it seems to work like this.

**Comment by fubar-coder** [comment]
That'll work. 4.0 will contain some big changes, but this version will take some time.

**Comment by rhegner** [comment]
I just came back to my old ticket. In the meantime, MySqlConnector support works out of the box!
